### DISCUSSION TITLE
Landsat sensor harmonization

### DISCUSSION THREAD

**Discussion by jobelanger** [worldbank/GEE_Zonal#3] [Q&A]
Hi All, I've been working to merge image collections from Landsat 5, 7 and 8 into one. There are two ways to go about this, as discussed via email with @andresfchamorro  -

1. Merge raw image collections and then calculate bands on the merged collection. Code [here](https://code.earthengine.google.com/4cd44815aba4a76b95b86a0f0b9e2874). (currently incomplete/with errors).
2. Calculate bands for each sensor and then merge collections together. Code [here](https://code.earthengine.google.com/b2311deb342f290ef216b4ee34ed652f).

I was able to successfully complete #2 for Landsat 5, 7, and 8, including with a cloud mask and gap fill algorithm. This was adapted to GEE from a USGS algorithm adapted by Noel Gorelick and Nicholas Clinton (code [here](https://code.earthengine.google.com/d20cba5268ccbe117e2fc1c5fefc33f3) and [here](https://stackoverflow.com/a/55271390/12820888)). I recommend checking the gapfill kernel size and cloud mask parameters for your use case. For my implementation in Afghanistan, these two significantly limit results but are preserved for data integrity. 

I had trouble completing #1, which involves renaming Landsat 8 bands to match those of L5 and L7. While methodology #2 works for my use case, I imagine raw merging to be potentially more computationally efficient and easier to work with, especially if created as an EE asset for later use. 

Welcoming any feedback or suggestions to the above. thanks!

**Comment by andresfchamorro** [comment]
Hi @jobelanger, this is great! The implementation for #2 makes sense. Wanted to ask if the statistics were somewhat consistent across landsat sensors? I just did something similar from the NDVI datasets and noticing the values are not very comparable (by looking at NDVI values for the same year but different Landsat collections). I am going to check to see if following this tutorial helps with harmonization: 
https://developers.google.com/earth-engine/tutorials/community/landsat-etm-to-oli-harmonization#references

**Reply by jobelanger** [reply]
thanks @andresfchamorro, I didn't get around to checking sensor comparability as I had to temporarily shelf this task. Please let me know if the tutorial rectifies this issue for you!

**Reply by jobelanger** [reply]
hey @andresfchamorro , finally picking this back up. did you end up using the GEE tutorial or getting harmonization to work in another way?
